### PR TITLE
feat: add config option to merge chromeOptions

### PR DESCRIPTION
**Pull request by amkirwan**
- merge desiredCapabilities.chromeOptions.args by default if specified
in config
- allow override of desiredCapabilities.chromeOptions.args with merge:
false

### CODE REVIEW

**Comment by dbushong** [comment]
Thanks for this feature.  A few observations:

1. Since `chromeOptions` has a lot of different things in it, I'd prefer something like `mergeArgs` to show what it's operating on
1. To maintain backward-compatibility, it should default to being `false`, so the 2nd arg to
    `config.get('desiredCapabilities.chromeOptions.mergeArgs')` should be `false`, I believe

File: lib/processes.js [hunk 1]
```diff
@@ -103,17 +103,32 @@ function launchAllProcesses(config) {
...
+
+      if (config.has('desiredCapabilities.chromeOptions.args')) {
+        if (config.has('desiredCapabilities.chromeOptions.merge')
+          && config.get('desiredCapabilities.chromeOptions.merge') === false) {
+          args = config.get('desiredCapabilities.chromeOptions.args');
+        } else {
+          args = args.concat(config.get('desiredCapabilities.chromeOptions.args'))
+            .filter(function filterIt(opt, idx, arr) {
+              return arr.indexOf(opt) === idx;
+            });
```
Comment (dbushong) [inline lib/processes.js]: I think this might need to be a bit more sophisticated for cases like `--disk-cache-size=2`

Comment (amkirwan) [reply lib/processes.js]: yeah that is a good point I forgot about options like `--disk-cache-size=2`

File: lib/processes.js [hunk 2]
```diff
@@ -103,17 +118,29 @@ function launchAllProcesses(config) {
...
-      ]);
+      ];
+
       if (config.get('chrome.headless', !process.env.DISPLAY)) {
         args.push('--headless');
       }
+
+      if (config.has('desiredCapabilities.chromeOptions.args')) {
+        if (config.has('desiredCapabilities.chromeOptions.mergeArgs')
+          && config.get('desiredCapabilities.chromeOptions.mergeArgs') === true) {
```
Comment (dbushong) [inline lib/processes.js]: can probably collapse this to:
```js
if (config.get('desiredCapabilities.chromeOptions.mergeArgs', false) === true) {
```

File: lib/processes.js [hunk 3]
```diff
@@ -61,6 +61,21 @@ function isChromeInDocker(browserName) {
...
+function mergeChromeOptions(baseOpts, configOpts) {
+  var reg = /^([-\w]+)=?([\.,\/\w]*)/;
+
+  var objOpts = baseOpts.concat(configOpts).reduce(function filterOpts(acc, val) {
+    var match = reg.exec(val);
+    acc[match[1]] = match[2];
+    return acc;
+  }, {});
+
+  return Object.entries(objOpts).reduce(function mergeOpts(acc, opt) {
```
Comment (dbushong) [inline lib/processes.js]: `Object.entries()` requires node 8+; at the moment we want to maintain compatibility with 6+; you can use `Object.keys()` and a bit more work and maintain compatibility

**Comment by dbushong** [comment]
LGTM; thanks!

**Pull request merged**